cfme/tests/test_rest.py

```diff
@@ -1261,3 +1263,52 @@ def _finalize():
     # fetch the latest product_info
     branding_info = api.get(href)["branding_info"]
     assert branding_info[image_type] == expected_name.format(image_type)
+
+
+@pytest.fixture
+def second_provider(appliance, request, provider):
+    # get and create second provider
+    providers_list = list_providers(
+        filters=[
+            ProviderFilter(
+                classes=[RHEVMProvider, VMwareProvider], required_tags=["default"]
+            )
+        ]
+    )
+    providers_list.remove(provider)
+    provider_2 = providers_list[0]
+    provider_2.create_rest()
+    request.addfinalizer(provider_2.delete_rest)
+
+    # wait until the data loads for the second provider
+    wait_for(
+        lambda: len(provider_2.rest_api_entity.vms.all) > 0,
+        fail_func=provider_2.refresh_provider_relationships,
+        timeout=300,
+        delay=5,
+    )
+    return provider_2
```

Comment:
Why are you going through this process instead of using a second provider marker?

The provider marker, and `setup_provider` fixture, adds providers via REST, validates their inventory, waits for refresh, accounts for 'problematic' providers that have already failed being added 3 times, and accounts for provider limits.

Reply:
If its not something you've seen, it'll look like:
```
@pytest.mark.provider([RHEVMProvider, VMwareProvider], required_tags=["default"], fixture_name='second_provider')
```

Reply:
Yes, thank you! This is new!